File: dataset/avatar/Python/Code/atcoder_AGC027_A.py
na = lambda : list ( map ( int , input ( ).split ( ) ) )
n , x = na ( )
a = sorted ( na ( ) )
ans = 0
if a [ 0 ] > x :
    print ( 0 )
    exit ( )
for i in a :
    if i > x :
        x = 0
        break
    x -= i
    ans += 1
if x != 0 :
    ans -= 1
print ( ans )

File: output/gpt_4o/avatar/algo_based_translation/python/java/atcoder_AGC027_A.java

import java.util.Arrays;
import java.util.Scanner;

public class atcoder_AGC027_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        // Lambda function equivalent in Java
        int[] na() {
            return Arrays.stream(scanner.nextLine().split(" "))
                         .mapToInt(Integer::parseInt)
                         .toArray();
        }

        int[] firstInput = na();
        int n = firstInput[0];
        int x = firstInput[1];

        int[] a = na();
        Arrays.sort(a);

        int ans = 0;

        if (a[0] > x) {
            System.out.println(0);
            return;
        }

        for (int i : a) {
            if (i > x) {
                x = 0;
                break;
            }
            x -= i;
            ans++;
        }

        if (x != 0) {
            ans--;
        }

        System.out.println(ans);
    }
}

